azurelinuxagent/ga/exthandlers.py

```diff
@@ -310,11 +311,74 @@ def handle_ext_handlers(self, etag=None):
                 logger.info("Extension handling is on hold")
                 return
 
+        wait_until = datetime.datetime.utcnow() + datetime.timedelta(minutes=DEFAULT_EXT_TIMEOUT_MINUTES)
+        prev_handler = None
+        ext_seq_required = self.is_ext_seq_required()
+
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            if ext_seq_required:
+                # Make sure that the previous extension is handled.
+                # If handled successfully, proceed with the current handler.
+                # Otherwise, skip the rest of them.
+                dep_level = ext_handler.sort_key()
+                if dep_level >= 0:
+                    if self.wait_for_prev_handler_completion(prev_handler, wait_until):
+                        prev_handler = ext_handler
+                    else:
+                        break
+
             self.handle_ext_handler(ext_handler, etag)
-    
+
+    def is_ext_seq_required(self):
+        '''
+        If all the extensions have same dependency level
+        It is considered as extension sequencing is not required
+        '''
+        dep_level = None
+        for ext_handler in self.ext_handlers.extHandlers:
+            if dep_level is None:
+                dep_level = ext_handler.sort_key()
+                continue
+
+            if dep_level != ext_handler.sort_key():
+                return True
+
+        return False
+
+    def wait_for_prev_handler_completion(self, prev_handler, wait_until):
```

Comment:
I would remove 'prev_' from the name of this function and its parameters. The function just waits for a handler, by itself it doesn't have a concept of "previous handler"